### PR TITLE
Enable int64 support for MatMul

### LINKED ISSUE #21241: Feature request: matmul operation for int64 datatype
For an open source project on [sMPC](https://en.wikipedia.org/wiki/Secure_multi-party_computation) in Tensorflow I am using the int64 datatype [to represent decimal numbers](https://www1.cs.fau.de/filepool/publications/octavian_securescm/secfp-fc10.pdf). However, the matmul operations is currently not supported for this datatype. Is there any chance this could be enabled in the future? Currently we are using the [Chinese remainder theorem](https://en.wikipedia.org/wiki/Chinese_remainder_theorem) to represent the numbers, but as a result all operations are significantly slower.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.3
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #21241 where there were no int64 support for MatMul. This fix adds int64 support for MatMul.

This fix fixes #21241.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by rmlarsen — APPROVED** [review]

File: tensorflow/python/kernel_tests/matmul_op_test.py [hunk 1]
```diff
@@ -214,9 +214,9 @@ def testInfixMatmulDoesDotProduct(self):
   sizes = [1, 3, 5]
   trans_options = [[False, False], [True, False], [False, True]]
   for use_static_shape in [False, True]:
-    for dtype in (np.int32, np.float16, np.float32, np.float64, np.complex64,
-                  np.complex128):
-      if not use_static_shape and dtype == np.int32:
+    for dtype in (np.int32, np.int64, np.float16, np.float32, np.float64,
+                  np.complex64, np.complex128):
+      if not use_static_shape and dtype == np.int32 and dtype == np.int64:
```
Comment (rmlarsen) [inline tensorflow/python/kernel_tests/matmul_op_test.py]: I think you meant
```
if not use_static_shape and (dtype == np.int32 or dtype == np.int64):
```

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by yongtang** [comment]
Thanks @rmlarsen for the review and sorry for the bug in the test. The PR has been updated. Please take a look.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 20 days with no activity and the `awaiting review` label has been applied.

**Comment by justin1121** [comment]
Any update on getting this merged? Very interested in seeing it land!

**Review by rmlarsen — APPROVED** [review]

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Pull request merged**